bundles/org.openhab.binding.boschspexor/src/main/java/org/openhab/binding/boschspexor/internal/BoschSpexorThingHandler.java

```diff
@@ -0,0 +1,369 @@
+    private static final String CHANNEL_ID_PROFILE_NAME = "profileName";
+    private static final String CHANNEL_ID_PROFILE_TYPE = "profileType";
+    private static final long DEFAULT_CACHE_TIMEOUT = 30;
+
+    private final Logger logger = LoggerFactory.getLogger(BoschSpexorThingHandler.class);
+
+    private Optional<ScheduledFuture<?>> pollEvent;
+    private ExpiringCache<SpexorInfo> cache;
+
+    public BoschSpexorThingHandler(Thing thing) {
+        super(thing);
+        this.pollEvent = Optional.empty();
+        cache = new ExpiringCache<SpexorInfo>(Duration.ofSeconds(DEFAULT_CACHE_TIMEOUT), this::getSpexorInfo);
+        if (logger.isDebugEnabled()) {
+            logger.debug("Bosch spexor handler was created");
+        }
+    }
+
+    private Optional<String> getSpexorID() {
+        return Optional.ofNullable((String) thing.getConfiguration().get(PROPERTY_SPEXOR_ID));
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (logger.isDebugEnabled()) {
+            logger.debug("received ChannelUID {} with command {}", channelUID, command);
+        }
+
+        if (getSpexorID().isEmpty()) {
+            logger.warn("thing is not well created and can't be used");
+            return;
+        }
+
+        if (CHANNEL_BOSCH_SPEXOR.equals(channelUID.getId())) {
+            if (command instanceof RefreshType) {
+                refreshThing();
+            }
+        } else if (GROUP_ID_OBSERVATIONS.equals(channelUID.getGroupId())) {
+            SpexorAPIService apiService = getSpexorAPIService();
+            if (command instanceof StringType) {
+                if (apiService == null) {
+                    logger.warn("spexor API service is not available and comand won't be performed");
+                    return;
+                }
+                SensorMode mode = null;
+                try {
+                    mode = SensorMode.valueOf(command.toString().toUpperCase());
+                } catch (IllegalArgumentException e) {
+                    logger.warn("no SensorMode could be found for command '{}'", command, e);
+                }
+                String type = channelUID.getIdWithoutGroup();
+                if (SensorMode.ACTIVATED.equals(mode) || SensorMode.DEACTIVATED.equals(mode)) {
+                    ObservationChangeStatus newObservationState = apiService.setObservation(getSpexorID().get(), type,
+                            SensorMode.ACTIVATED.equals(mode));
+                    if (logger.isDebugEnabled()) {
+                        logger.debug("setting new observation state for {} to {} was {}", type, mode,
+                                newObservationState.getStatusCode());
+                    }
+                    updateState(channelUID, new StringType(newObservationState.getSensorMode().name()));
+                } else {
+                    updateStatus(ThingStatus.ONLINE, ThingStatusDetail.CONFIGURATION_ERROR,
```

Comment:
that behavior is still correct - device is online but not well configured.

Reply:
If the binding would continue to work as before when this happens, the state is still just online. It looks like you want to inform the user the command given was not valid. The way to inform the user is to set the state of the channel to original value or undef or make a specific state unknown and set the channel to that state. Also log the error message. See it from an other point. The state reflects the state of the binding with the device. If openHAB would restart it needs to determine the state again. In that case it will just show online and if it would have had the configuration error state that information would be lost.

Reply:
changed it to the undefined proposal and added an error message